test/suites/device-onvif-camera/main_test.go

```diff
@@ -0,0 +1,88 @@
+package test
+
+import (
+	"edgex-snap-testing/test/utils"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	deviceOnvifcameraSnap        = "edgex-device-onvif-camera"
+	deviceOnvifCameraApp         = "device-onvif-camera"
+	deviceOnvifcameraService     = deviceOnvifcameraSnap + "." + deviceOnvifCameraApp
+	deviceOnvifcameraServicePort = "59984"
+	testOnvifCredFile            = "/var/snap/edgex-device-onvif-camera/current/config/device-onvif-camera/test-onvif-credentials.json"
+)
+
+func TestMain(m *testing.M) {
+
+	log.Println("[SETUP]")
+	start := time.Now()
+	// start clean
+	utils.SnapRemove(nil,
+		deviceOnvifcameraSnap,
+		"edgexfoundry",
+	)
+
+	// install the device snap before edgexfoundry
+	// to catch build error sooner and stop
+	if utils.LocalSnap != "" {
+		utils.SnapInstallFromFile(nil, utils.LocalSnap)
+	} else {
+		utils.SnapInstallFromStore(nil, deviceOnvifcameraSnap, utils.ServiceChannel)
+	}
+	utils.SnapInstallFromStore(nil, "edgexfoundry", utils.PlatformChannel)
+
+	// make sure all services are online before starting the tests
+	utils.WaitPlatformOnline(nil)
+
+	// for local build, the interface isn't auto-connected.
+	// connect manually regardless
+	utils.SnapConnect(nil,
+		"edgexfoundry:edgex-secretstore-token",
+		deviceOnvifcameraSnap+":edgex-secretstore-token",
+	)
+
+	log.Printf("Copying test credentials to %s", testOnvifCredFile)
+	utils.Exec(nil, "sudo cp test-onvif-credentials.json "+testOnvifCredFile)
+
+	utils.SnapSet(nil, deviceOnvifcameraSnap, "config.secretstore-secretsfile", testOnvifCredFile)
```

Comment:
This setup step depends on the working snap options which is actually tested afterwards. Let's try to make the service run without any secrets. This shouldn't be mandatory as not all ONVIF cameras will have authentication enabled.

Maybe setting the value to empty string will do the trick? https://github.com/edgexfoundry/device-onvif-camera/blob/main/cmd/res/configuration.toml#L113

Reply:
Make sense.  setting `DefaultSecretPath`'s value to an empty string or keeping the default value all works. The onvif-camera service will start without error.

Is this nesscessary to override `DefaultSecretPath=""`in snapcraft?